File: sample-ip-echo-service/ip-echo.py
from flask import Flask, request
import os

app = Flask(__name__)

port = int(os.environ.get('NODE_PORT', 3000))

@app.route('/')
def get_ip():
    """
    Returns the client's IP address in plain text.
    """
    ip_address = request.remote_addr
    return ip_address, 200, {'Content-Type': 'text/plain'}

if __name__ == '__main__':
    app.run(host='0.0.0.0', port=port)

File: sample-ip-echo-service/ip-echo.js
const express = require('express')

const app = express()

const port = process.env.NODE_PORT || 3000

app.get( '/' , ( req , res ) => {
    res.statusCode = 200
    res.setHeader('Content-Type', 'text/plain')
    res.write( req.ip )
    res.send()
})

app.listen( port , '0.0.0.0' )
